test/sql/test_function/R/test_translate

```diff
@@ -0,0 +1,27 @@
+-- name: test translate
+select translate('abcdefg', 'abcdefg', '');
```

Comment:
Add some test cases:
- which contains non-ASCII characters.
- or some parameters are invalid, such as non-constant or non-varchar `from_string` and `to_string`